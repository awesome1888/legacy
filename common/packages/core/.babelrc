{
  "presets": [
    "@babel/react",
    [
      "@babel/env",
      {
        "modules": "commonjs",
        "targets": {
          "browsers": ["last 2 versions"]
        }
      }
    ],
    "@babel/preset-typescript"
  ],
  "plugins": [
    [
      "@babel/plugin-proposal-decorators",
      { "legacy": true }
    ],
    "@babel/plugin-proposal-object-rest-spread",
    [
      "@babel/plugin-proposal-class-properties",
      { "loose": true }
    ]
  ]
}
