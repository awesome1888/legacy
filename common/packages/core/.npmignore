demo
.idea
*.log
*.tar.gz
jest
.eslint*
.gitkeep
.babelrc
script
