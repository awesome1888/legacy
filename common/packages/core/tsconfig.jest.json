{
    "compilerOptions": {
        "module": "CommonJS",
        "target": "es5",
        "allowJs": true,
        "skipLibCheck": false,
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "strict": true,
        "forceConsistentCasingInFileNames": true,
        "moduleResolution": "node",
        "resolveJsonModule": true,
        "isolatedModules": false,
        "experimentalDecorators": true,
        "lib": ["esnext.asynciterable", "esnext", "es2017", "dom"],
        "removeComments": true,
        "emitDecoratorMetadata": true,
        "baseUrl": "./src/",
        "declaration": false,
        "types": []
    },
    "exclude": ["node_modules", "build"],
    "include": ["src/**/*", "src/global.d.ts"]
}
