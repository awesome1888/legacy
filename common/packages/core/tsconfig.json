{
    "compilerOptions": {
        "module": "CommonJS",
        "target": "es2018",
        "allowJs": false,
        "skipLibCheck": false,
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "strict": true,
        "forceConsistentCasingInFileNames": true,
        "moduleResolution": "node",
        "resolveJsonModule": true,
        "isolatedModules": false,
        "experimentalDecorators": true,
        "lib": ["esnext.asynciterable", "esnext", "es2017", "dom"],
        "removeComments": true,
        "emitDecoratorMetadata": true,
        "sourceMap": true,
        "baseUrl": "./src/",
        "outDir": "./build/",
        "declaration": true,
        "types": []
    },
    "exclude": ["node_modules", "build"],
    "include": ["src/**/*", "src/global.d.ts"]
}
