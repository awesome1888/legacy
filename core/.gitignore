node_modules/
*.log
*.tar.gz
.idea/
.yarn-error.log
