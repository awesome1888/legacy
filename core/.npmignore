demo
.idea
*.log
*.tar.gz
