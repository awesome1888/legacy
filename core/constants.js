export const DB_TABLE_PREFIX = 'eq_';
export const DB_REF_TABLE_PREFIX = `${DB_TABLE_PREFIX}ref_`;
export const DB_ENTITY_TABLE_PREFIX = `${DB_TABLE_PREFIX}e_`;
export const DB_MIGRATION_TABLE_NAME = `${DB_TABLE_PREFIX}migrations`;
export const DB_SCHEMA_TABLE_NAME = `${DB_TABLE_PREFIX}schema`;
export const DB_IDENTIFIER_LENGTH = 63;
export const DB_VARCHAR_DEF_LENGTH = 255;
export const DB_CODE_COLUMN_LENGTH = 36;
export const QUERY_FIND_MAX_PAGE_SIZE = 50;

export const ENTITY_USER_NAME = 'user';
export const ENTITY_GROUP_NAME = 'group';
export const ENTITY_CODE_FIELD_NAME = 'code';

/** @deprecated */
export const ENTITY_TYPE_STRING = 'string';
/** @deprecated */
export const ENTITY_TYPE_NUMBER = 'number';
/** @deprecated */
export const ENTITY_TYPE_BOOLEAN = 'boolean';
/** @deprecated */
export const ENTITY_TYPE_DATE = 'datetime';
