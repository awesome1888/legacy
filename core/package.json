{
    "name": "project-minimum-core",
    "version": "1.0.0",
    "description": "Some core stuff for the Project Minimum microservices",
    "main": "index.js",
    "browser": {
        "./build/server.js": "./build/client.js"
    },
    "scripts": {
        "start": "npm-run-all --parallel build.*.dev",
        "build": "npm-run-all --parallel build.*.prod",
        "build.server.prod": "webpack --config webpack.server.js --mode production",
        "build.client.prod": "webpack --config webpack.client.js --mode production",
        "build.dev": "npm-run-all --parallel build.*.dev",
        "build.server.dev": "webpack --config webpack.server.js --mode development --watch",
        "build.client.dev": "webpack --config webpack.client.js --mode development --watch",
        "test.u": "jest -c ./jest/config.unit.js --forceExit"
    },
    "keywords": [],
    "author": "gannochenko.sv@gmail.com",
    "license": "MIT",
    "repository": {
        "type": "git",
        "url": "https://github.com/awesome1888/project-minimum-core"
    },
    "devDependencies": {
        "@babel/core": "^7.4.4",
        "@babel/plugin-proposal-class-properties": "^7.4.4",
        "@babel/plugin-proposal-object-rest-spread": "^7.4.4",
        "@babel/preset-env": "^7.4.4",
        "@babel/preset-react": "^7.0.0",
        "babel-loader": "^8.0.6",
        "babel-plugin-styled-components": "^1.10.0",
        "husky": "^2.3.0",
        "jest": "^24.8.0",
        "npm-run-all": "^4.1.5",
        "prettier": "^1.17.1",
        "pretty-quick": "^1.10.0",
        "webpack": "^4.32.0",
        "webpack-cli": "^3.3.2",
        "webpack-node-externals": "^1.7.2"
    },
    "dependencies": {},
    "peerDependencies": {
        "react": "16.x"
    },
    "husky": {
        "hooks": {
            "pre-commit": "pretty-quick --staged"
        }
    }
}
