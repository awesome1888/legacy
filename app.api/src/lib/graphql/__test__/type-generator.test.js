import { Schema } from '@project-minimum/core';
import TypeGenerator from '../type-generator';
import schemaJSON from '../../../__test__/schema';

describe('GQL Type generator', () => {
    // beforeAll(async () => {
    // });
    // beforeEach(async () => {
    // });
    describe('make()', () => {
        it('should generate gql type declaration', async () => {
            const schema = new Schema({
                schema: schemaJSON,
                draft: true,
                version: 2,
            });
            const gql = TypeGenerator.make(schema);
            expect(gql).toMatchSnapshot();
        });

        it('should handle empty schema correctly', async () => {
            const schema = new Schema({});
            const gql = TypeGenerator.make(schema);
            expect(gql).toEqual([]);
        });
    });
});
