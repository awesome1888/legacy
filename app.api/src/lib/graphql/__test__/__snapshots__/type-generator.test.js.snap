// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GQL Type generator Should generate gql type declaration 1`] = `
Array [
  "
type ImportantPersonResult {
    errors: [Error]
    data: ImportantPerson
}

type ImportantPersonSearchResult {
    errors: [Error]
    data: [ImportantPerson]!
    limit: Int!
    offset: Int!
    count: Int
}

type ImportantPerson {
    code: String
    full_name: String
    tags: [String]
    lucky_numbers: [Int]
    birth_date: String
    has_pets: Boolean
    pets(
                filter: IPetFilter
                sort: IPetSort
                limit: Int
                offset: Int
                page: Int
                pageSize: Int
                count: Int
            ): [Pet]
    tools(
                filter: IToolFilter
                sort: IToolSort
                limit: Int
                offset: Int
                page: Int
                pageSize: Int
                count: Int
            ): [Tool]
    partner: ImportantPerson
}

input IImportantPerson {
    full_name: String
    tags: [String]
    lucky_numbers: [Int]
    birth_date: String
    has_pets: Boolean
    pets: [String]
    tools: [String]
    partner: String
}

input IImportantPersonFilter {
    SYSLogic: FilterLogic
    SYSSubFilter: [IImportantPersonFilter]
    code: IFilterFieldValue
    full_name: IFilterFieldValue
    tags: IFilterFieldValue
    lucky_numbers: IFilterFieldValue
    birth_date: IFilterFieldValue
    has_pets: IFilterFieldValue
    pets: IFilterFieldValue
    tools: IFilterFieldValue
    partner: IFilterFieldValue
}

input IImportantPersonSort {
    code: SortOrder
    full_name: SortOrder
    tags: SortOrder
    lucky_numbers: SortOrder
    birth_date: SortOrder
    has_pets: SortOrder
    pets: SortOrder
    tools: SortOrder
    partner: SortOrder
}

type Query {
    ImportantPersonGet(code: String!): ImportantPersonResult
    ImportantPersonFind(
        filter: IImportantPersonFilter
        search: String
        sort: IImportantPersonSort
        limit: Int
        offset: Int
        page: Int
        pageSize: Int
        count: Boolean
    ): ImportantPersonSearchResult
}

type Mutation {
    ImportantPersonDelete(code: String!): ImportantPersonResult
    ImportantPersonPut(code: String, data: IImportantPerson!): ImportantPersonResult
}
        ",
  "
type PetResult {
    errors: [Error]
    data: Pet
}

type PetSearchResult {
    errors: [Error]
    data: [Pet]!
    limit: Int!
    offset: Int!
    count: Int
}

type Pet {
    code: String
    nickname: String
}

input IPet {
    nickname: String
}

input IPetFilter {
    SYSLogic: FilterLogic
    SYSSubFilter: [IPetFilter]
    code: IFilterFieldValue
    nickname: IFilterFieldValue
}

input IPetSort {
    code: SortOrder
    nickname: SortOrder
}

type Query {
    PetGet(code: String!): PetResult
    PetFind(
        filter: IPetFilter
        search: String
        sort: IPetSort
        limit: Int
        offset: Int
        page: Int
        pageSize: Int
        count: Boolean
    ): PetSearchResult
}

type Mutation {
    PetDelete(code: String!): PetResult
    PetPut(code: String, data: IPet!): PetResult
}
        ",
  "
type ToolResult {
    errors: [Error]
    data: Tool
}

type ToolSearchResult {
    errors: [Error]
    data: [Tool]!
    limit: Int!
    offset: Int!
    count: Int
}

type Tool {
    code: String
    name: String
}

input ITool {
    name: String
}

input IToolFilter {
    SYSLogic: FilterLogic
    SYSSubFilter: [IToolFilter]
    code: IFilterFieldValue
    name: IFilterFieldValue
}

input IToolSort {
    code: SortOrder
    name: SortOrder
}

type Query {
    ToolGet(code: String!): ToolResult
    ToolFind(
        filter: IToolFilter
        search: String
        sort: IToolSort
        limit: Int
        offset: Int
        page: Int
        pageSize: Int
        count: Boolean
    ): ToolSearchResult
}

type Mutation {
    ToolDelete(code: String!): ToolResult
    ToolPut(code: String, data: ITool!): ToolResult
}
        ",
]
`;
